DIVA-DIA/DIVA-DAF
# Add strict parameter

### Description
Make it possible to define in the config if we load a model in a strict fashion or not ([description](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.load_state_dict)). If the model is loaded in an non-strict fashion and the two models have differences we are logging the missed keys (keys wich are in the model but not in the loaded state_dict) and unexpected keys (keys which are in the loaded state_dicts but not in the model). Care: #5 must be merged first. close #15 

### How to Test/Run?
If you want to see the printed out strict parameter:
`python run.py experiment=development_baby_unet_cb55_10 trainer.max_epochs=3 test=false +model.backbone.strict=False`

If you want to test the printouts:
`python run.py experiment=development_baby_unet_cb55_10 trainer.max_epochs=3 test=false +model.backbone.strict=False +model.backbone.path_to_weights=/PATH/TO/TRAINED/PTH/`

### Something missing?
Should not
